### ISSUE TITLE
Single-lead downstream task

### ISSUE THREAD

**Issue by fastlib** [Edoar-do/HuBERT-ECG#5]
Thanks for your amazing work and the fact you open-sourced everything! This is great for the research community! 

I'm currently working on a benchmark that compares different machine learning methods for the task of single-lead ECG classification. Your model works with 12 leads. What is your best guess on how to adapt the model to work with single-lead (lead II) data? Using only lead II and fill the other leads with zeros? Copy lead II to all other leads or change the first layer to a 1d layer and average all its pretrained weights? What do you think?

**Comment by Edoar-do** [comment]
Hello @fastlib!

Honestly, I have no idea of what could do the job. You should try several configurations😅 
One could think of:
- using a single lead, placed in the right spot (see fig.1 to see the lead order), filling the rest of the input signal with 0s and compute the attention mask accordingly;
- to pick a lead and replicate it 12 times to match the expected signal shape
- definitely **not** averaging model weights

In any case, I think you're gonna introduce a distribution shift, the impacy of which I cannot tell in advance, that would require an additional training step to be fully absorbed. This, however, could result in losing or decreasing the capacity of processing 12-lead ECGs.

All the above points are just ideas, speculations and/or fears. Please be aware of that
